spec.html

```diff
@@ -171,6 +172,19 @@ <h1>Web Scripting</h1>
     <p>Each Web browser and server that supports ECMAScript supplies its own host environment, completing the ECMAScript execution environment.</p>
   </emu-clause>
 
+  <emu-clause id="sec-hosts-and-implementations">
+    <h1>Hosts and Implementations</h1>
+    <p>To aid integrating ECMAScript into host environments, this specification defers the definition certain facilities (e.g., abstract operations), either in whole or in part, to a source outside of this specification. Editorially, this specification distinguishes the following kinds of deferrals.</p>
+    <p>An <dfn id="sec-implementation">implementation</dfn> is an external source that further defines facilities enumerated in Annex <emu-xref href="#sec-host-layering-points"></emu-xref> or those that are marked as implementation-defined or implementation-approximated. In informal use, an implementation refers to a concrete artefact, such as a particular web browser.</p>
+    <p>An <dfn id="sec-implementation-defined">implementation-defined</dfn> facility is one that defers its definition to an external source without further qualification. This specification does not make any recommendations for particular behaviours, and conforming implementations are free to choose any behaviour within the constraints put forth by this specification.</p>
+    <p>An <dfn id="sec-implementation-approximated">implementation-approximated</dfn> facility is one that defers its definition to an external source while recommending an ideal behaviour. While conforming implementations are free to choose any behaviour within the constraints put forth by this specification, they are encouraged to strive to approximate the ideal. Some mathematical operations, such as <emu-xref="#sec-math.exp" title></emu-xref>, are implementation-approximated.</p>
+    <p>A <dfn id="sec-host">host</dfn> is an external source that further defines facilities listed in Annex <emu-xref href="#sec-host-layering-points"></emu-xref> but does not further define other implementation-defined or implementation-approximated facilities. In informal use, a host refers to the set of all implementations, such as the set of all web browsers, that interface with this specification in the same way via Annex <emu-xref href="#sec-host-layering-points"></emu-xref>. A host is often an external specification, such as WHATWG HTML. In other words, facilities that are host-defined are often further defined in external specifications.</p>
```

Comment:
"WHATWG HTML" seems like it should be linked, maybe?

Reply:
Will add.